stl/inc/random

```diff
@@ -2929,14 +2929,17 @@ private:
             _Res   = _Xx2;
             _Valid = false;
         } else { // generate two values, store one, return one
+            const _Ty _MinSx{static_cast<_Ty>(7.8159598771420286e-306)};

```

Comment:
The issue with this is that the exact value depends on the type and, more importantly, how `log` is calculated. I was thinking of using scaling, similar to #935 for small arguments to `log`. Something like the following (caveats: not well-tested, only `double`, not STL style). The 1e-4 threshold is mostly arbitrary, but I could make it more rigorous if people are interested. Or not, since it should only take that path once every 10^8 calls.
```
double Vx1, Vx2, Fx;
for (;;) {
    Vx1 = 2 * std::generate_canonical<double, size_t(-1)>(g) - 1;
    Vx2 = 2 * std::generate_canonical<double, size_t(-1)>(g) - 1;
    double Sx = Vx1 * Vx1 + Vx2 * Vx2;
    if (Sx >= 1.0 || (Vx1 == 0.0 && Vx2 == 0.0)) {
        continue;
    }
    double tmp;
    if (Sx <= 1.0e-4) {
        double maxAbs = std::max(std::abs(Vx1), std::abs(Vx2));
        const int expMax = std::ilogb(maxAbs);
        Vx1 = std::scalbn(Vx1, -expMax);
        Vx2 = std::scalbn(Vx2, -expMax);
        Sx = Vx1 * Vx1 + Vx2 * Vx2;
        constexpr double ln2 = 0.6931471805599453;
        tmp = std::log(Sx) + static_cast<double>(expMax) * (2.0 * ln2);
    }
    else {
        tmp = std::log(Sx);
    }
    Fx = sqrt(-2.0 * tmp / Sx);
    break;
}
```

Reply:
@MattStephanson 
I just refactor your code in the most recent commit. I didn't encounter any bugs during my tests.
Thank you for the idea and the implementation.